ecl/hthor/hthor.cpp

```diff
@@ -10363,6 +10364,681 @@ void CHThorExternalActivity::stop()
+            WARNLOG("%s", buff.str());
+            agent.addWuException(buff.str(), WRN_SkipMissingOptFile, SeverityInformation, "hthor");
+        }
+    }
+
+    assertex(subfiles.ordinality() != 0);
+}
+
+void CHThorNewDiskReadBaseActivity::gatherInfo(IFileDescriptor * fileDesc)
+{
+    if (fileDesc)
+    {
+        if (!agent.queryResolveFilesLocally())
+        {
+            grouped = fileDesc->isGrouped();
+            if (grouped != ((helper.getFlags() & TDXgrouped) != 0))
+            {
+                StringBuffer msg;
+                msg.append("DFS and code generated group info. differs: DFS(").append(grouped ? "grouped" : "ungrouped").append("), CodeGen(").append(grouped ? "ungrouped" : "grouped").append("), using DFS info");
+                WARNLOG("%s", msg.str());
+                agent.addWuException(msg.str(), WRN_MismatchGroupInfo, SeverityError, "hthor");
+            }
+        }
+        else
+            grouped = ((helper.getFlags() & TDXgrouped) != 0);
+    }
+    else
+    {
+        grouped = ((helper.getFlags() & TDXgrouped) != 0);
+    }
+}
+
+CHThorNewDiskReadBaseActivity::InputFileInfo * CHThorNewDiskReadBaseActivity::extractFileInformation(IDistributedFile * distributedFile)
+{
+    Owned<IPropertyTree> meta = createPTree();
+    unsigned actualCrc = helper.getDiskFormatCrc();
+    Linked<IOutputMetaData> actualDiskMeta = expectedDiskMeta;
+    bool compressed = false;
+    bool blockcompressed = false;
+
+    if (distributedFile)
+    {
+        //MORE:
+        //const char *kind = queryFileKind(dFile);
+        //if (strisame(kind, "flat") || (RecordTranslationMode::AlwaysDisk == translationMode))
+
+        //Yuk this will be horrible - it needs to cache it for each distributed file
+        //and also common them up if they are the same.
+        IPropertyTree &props = distributedFile->queryAttributes();
+        Owned<IOutputMetaData> publishedMeta = getDaliLayoutInfo(props);
+        if (publishedMeta)
+        {
+            actualDiskMeta.setown(publishedMeta.getClear());
+            actualCrc = props.getPropInt("@formatCrc");
+        }
+
+        /*
+         * If a file part can be accessed local, then read it locally
+         * If a file part supports a remote stream, then use that
+         * Otherwise failover to the legacy remote access.
+         */
+        compressed = distributedFile->isCompressed(&blockcompressed); //try new decompression, fall back to old unless marked as block
+        size32_t dfsSize = props.getPropInt("@recordSize");
+        if (dfsSize != 0)
+            meta->setPropInt("dfsRecordSize", dfsSize);
+
+        //Check for encrption key
+        void *k;
+        size32_t kl;
+        helper.getEncryptKey(kl,k);
+        if (kl)
+        {
+            meta->setPropBin("encryptionKey", kl, k);
+            blockcompressed = true;
+            compressed = true;
+        }
+    }
+    else
```

Comment:
Do we need an empty else here? Should there be a comment in the else saying "No need to do X because Y"

Reply:
I'll look at it.